velox/common/memory/tests/MemoryPoolTest.cpp

```diff
@@ -524,6 +524,32 @@ TEST_P(MemoryPoolTest, grow) {
   leaf->free(buf, 1 * MB);
 }
 
+TEST_P(MemoryPoolTest, releasableMemory) {
+  struct TestParam {
+    int64_t usedBytes;
+    int64_t reservedBytes;
+  };
+  std::vector<TestParam> testParams{
+      {2345, 98760},
+      {1, 1024},
+      {4096, 4096},
+      {1 * MB, 16 * MB},
+      {6 * MB, 7 * MB},
+      {123 * MB, 200 * MB},
+      {100 * MB, 50 * MB}};
+  auto root = getMemoryManager()->addRootPool("releasableMemory", 4 * GB);
+  for (const auto& testParam : testParams) {
+    auto leaf = root->addLeafChild("leafPool");
+    leaf->maybeReserve(testParam.reservedBytes);
+    void* buffer = leaf->allocate(testParam.usedBytes);
+    const auto reservedBytes = leaf->reservedBytes();
+    const auto releasableBytes = leaf->releasableReservation();
```

Comment:
Can you add to test this API for non-leaf memory pools? Like root and aggregate? Thanks!